lib/templates/httpTemplate.ts

```diff
@@ -1,67 +1,65 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-const url = require('url'),
-  uuid = require('uuid');
-import utils = require('../util/utils');
+import * as url from 'url'
+import * as utils from '../util/utils'
 
-class HttpTemplate {
+export class HttpTemplate {
 
-  constructor(public request: any, public responses: any) {
+  constructor(public readonly request: any, public readonly responses: any) {
   }
 
-  getHost() {
-    let result = 'management.azure.com';
+  getHost(): string|undefined {
+    let result: string|undefined = 'management.azure.com'
     if (this.request.url) {
-      result = url.parse(this.request.url).host;
+      result = url.parse(this.request.url).host
     }
-    return result;
+    return result
   }
 
-  getCurlRequestHeaders(padding?: any) {
-    let result = ``;
-    if (!padding) padding = ``;
+  getCurlRequestHeaders(padding?: any): string {
+    let result = ``
+    if (!padding) padding = ``
     if (this.request.body) {
-      result += `\n${padding}-H 'Content-Length: ${JSON.stringify(this.request.body).length}' \\`;
+      result += `\n${padding}-H 'Content-Length: ${JSON.stringify(this.request.body).length}' \\`
     }
     if (this.request.headers) {
-      let headers = utils.getKeys(this.request.headers);
+      let headers = utils.getKeys(this.request.headers)
 
       for (let i = 0; i < headers.length; i++) {
-        let headerName = headers[i];
-        result += `\n${padding}-H '${headerName}: ${this.request.headers[headerName]}' \\`;
+        let headerName = headers[i]
+        result += `\n${padding}-H '${headerName}: ${this.request.headers[headerName]}' \\`
       }
     }
-    return result;
+    return result
   }
 
-  getRequestBody() {
-    let body = ``;
+  getRequestBody(): string {
+    let body = ``
```

Comment:
>dy = `` [](start = 10, length = 7)

no ; ?

Reply:
I would maybe install TSlint in vscode so it detects issues like this

---
In reply to: [192216095](https://github.com/Azure/oav/pull/254#discussion_r192216095) [](ancestors = 192216095)

Reply:
or even add a tslint.json file ( we dont need to enable it as part of build yet) and we can run something like tslint  -c tslint.json './src/**/*.ts*'

---
In reply to: [192216349](https://github.com/Azure/oav/pull/254#discussion_r192216349) [](ancestors = 192216349,192216095)